drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -725,7 +726,8 @@ __imr void adsp_mm_restore_context(void *storage_buffer)
 
 	while (phys_addr != 0) {
 		uint32_t phys_addr_uncached =
-				POINTER_TO_UINT(z_soc_uncached_ptr(UINT_TO_POINTER(phys_addr)));
+				POINTER_TO_UINT(z_soc_uncached_ptr(
+					(void __sparse_cache *)UINT_TO_POINTER(phys_addr)));
```

Comment:
At the risk of showing my ignorance:
```suggestion
					(__sparse_cache void *)UINT_TO_POINTER(phys_addr)));
```

For better readability and consistency.

Reply:
I was simply following what others have been doing.

Reply:
Well, consistency has to start somewhere :-)

@lyakh is there a technical reason for `__sparse_cache` and `__sparse_force` to be placed differently?

Reply:
> Well, consistency has to start somewhere :-)
> 
> @lyakh is there a technical reason for `__sparse_cache` and `__sparse_force` to be placed differently?

To quote @dcpleung "I was simply following what others have been doing" :-) And for me the example to follow was the kernel. Their analog of our `__sparse_force` is `__force` so you can grep kernel sources for that and see it's mostly used as the first item in type-cast statements. Although, I must say, not everywhere. As for `__sparse_cache` - look how, e.g. `__user` is placed in the kernel.